test/local_creds_test.go

```diff
@@ -27,13 +27,34 @@ import (
 	"time"
 
 	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials"
 	"google.golang.org/grpc/credentials/local"
+	"google.golang.org/grpc/peer"
+	"google.golang.org/grpc/status"
 	testpb "google.golang.org/grpc/test/grpc_testing"
 )
 
 func testE2ESucceed(network, address string) error {
 	ss := &stubServer{
-		emptyCall: func(context.Context, *testpb.Empty) (*testpb.Empty, error) {
+		emptyCall: func(ctx context.Context, in *testpb.Empty) (*testpb.Empty, error) {
+			pr, ok := peer.FromContext(ctx)
+			if !ok {
+				return nil, status.Error(codes.DataLoss, "Failed to get peer from ctx")
+			}
+			// Check security level
+			info := pr.AuthInfo.(local.Info)
+			secLevel := info.CommonAuthInfo.SecurityLevel
+			switch network {
+			case "unix":
+				if secLevel != credentials.PrivacyAndIntegrity {
+					return nil, status.Errorf(codes.Unauthenticated, "Wrong security level: got %q, want %q", secLevel.String(), credentials.PrivacyAndIntegrity.String())
```

Comment:
I am pretty sure `fmt.Sprintf` (which `status.Errorf` uses) will call `.String()` for you in this case (with `%q`) - won't it?

Reply:
Yes, you are right. Thanks for the suggestion. The code is updated.